File: 1.1/src/c/cz/mg/vulkan/PFN_vkResetDescriptorPool.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkResetDescriptorPool_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong descriptorPool, jlong flags, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkResetDescriptorPool f = (PFN_vkResetDescriptorPool)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkDescriptorPool*)jniLongToPointer(descriptorPool)),
        *((VkDescriptorPoolResetFlags*)jniLongToPointer(flags))
    );
}


jint Java_cz_mg_vulkan_PFNvkResetDescriptorPool_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong descriptorPool, jint flags){
    (void)env;
    (void)jc;
    PFN_vkResetDescriptorPool f = (PFN_vkResetDescriptorPool)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkDescriptorPool*)jniLongToPointer(descriptorPool)),
        (VkDescriptorPoolResetFlags)flags
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkResetDescriptorPool.java
package cz.mg.vulkan;

public class PFNvkResetDescriptorPool extends VkFunctionPointer {
    public PFNvkResetDescriptorPool() {
    }

    protected PFNvkResetDescriptorPool(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkResetDescriptorPool(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkResetDescriptorPool(long value) {
        setValue(value);
    }

    public PFNvkResetDescriptorPool(VkInstance instance) {
        super(instance, new VkString("vkResetDescriptorPool"));
    }

    public void call(VkDevice device, VkDescriptorPool descriptorPool, VkDescriptorPoolResetFlags flags, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), descriptorPool != null ? descriptorPool.getVkAddress() : VkPointer.getNullAddressNative(), flags != null ? flags.getVkAddress() : VkPointer.getNullAddressNative(), rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long descriptorPool, long flags, long rval);


    public int call(VkDevice device, VkDescriptorPool descriptorPool, int flags){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), descriptorPool != null ? descriptorPool.getVkAddress() : VkPointer.getNullAddressNative(), flags);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, long descriptorPool, int flags);

}
